lib/internal/Magento/Framework/Image/Adapter/ImageMagick.php

```diff
@@ -562,4 +580,42 @@ private function addSingleWatermark($positionX, int $positionY, \Imagick $waterm
             $compositeChannels
         );
     }
+
+    /**
+     * Get and store the image colorspace.
+     *
+     * @return int
+     */
+    public function getColorspace(): int
```

Comment:
Please, make method private

Reply:
@sidolov I can do that, but then this behavior cannot be tested like it is right now. It depends on these functions being public.

Reply:
I understand that but it's better to eliminate the possibility of inheritance here.

Reply:
Those tests are not very useful anyway. The "business" value it to have proper colors, not to get palette code properly